src/charm.py

```diff
@@ -528,12 +528,9 @@ def _create_resources(self) -> None:
                     logger.debug(f"created {str(resource)}")
         except ApiError as e:
             # The 409 error code means that the resource was already created
-            # or has a higher version. This can happen if Patroni creates a
-            # resource that the charm is expected to create.
-            if e.status.code == 409:
-                logger.debug("replacing resource: %s.", str(resource.to_dict()))
-                client.replace(resource)
-            else:
+            # or has a higher version. This can happen when multiple calls are
+            # made to this function.
+            if e.status.code != 409:
```

Comment:
In general, if we need to modify/change objects created by patroni - it is better to use the `client.apply(resource)` call.
So configuration will be merged by server-side apply.

Reply:
Thanks for the info, @delgod! I'll use that on [DPE-1470](https://warthogs.atlassian.net/browse/DPE-1470).